rasa/core/featurizers/single_state_featurizer.py

```diff
@@ -0,0 +1,244 @@
+            # its value
+            if state_feature in self._default_feature_states[attribute]:
+                features[self._default_feature_states[attribute][state_feature]] = value
+        features = np.expand_dims(features, 0)
+
+        if sparse:
+            features = scipy.sparse.coo_matrix(features)
+
+        features = Features(
+            features, FEATURE_TYPE_SENTENCE, attribute, self.__class__.__name__
+        )
+        return [features]
+
+    @staticmethod
+    def _to_sparse_sentence_features(
+        sparse_sequence_features: List["Features"], attribute: Text
+    ) -> List["Features"]:
+        sparse_sentence_features = []
+        for feature in sparse_sequence_features:
+            sparse_sentence_features.append(
+                Features(
+                    scipy.sparse.coo_matrix(feature.features.sum(0)),
+                    FEATURE_TYPE_SENTENCE,
+                    attribute,
+                    feature.origin,
+                )
+            )
+        return sparse_sentence_features
+
+    def _extract_state_features(
+        self,
+        sub_state: SubState,
+        interpreter: Optional[NaturalLanguageInterpreter],
+        sparse: bool = False,
+    ) -> Dict[Text, List["Features"]]:
+
+        output = defaultdict(list)
+        message = Message(data=sub_state)
+        # remove entities from possible attributes
+        attributes = set(
+            attribute for attribute in sub_state.keys() if attribute != ENTITIES
+        )
+
+        if interpreter is not None:
+            parsed_message = interpreter.synchronous_parse_message(message)
+            for attribute in attributes:
+                if parsed_message is not None:
+                    all_features = parsed_message.get_sparse_features(
+                        attribute
+                    ) + parsed_message.get_dense_features(attribute)
+                else:
+                    all_features = ()
+
+                for features in all_features:
+                    if features is not None:
+                        output[attribute].append(features)
+
+            # transform sequence sparse features to sentence sparse features
+            # for intent and action_name
+            if output.get(INTENT):
+                output[INTENT] = self._to_sparse_sentence_features(
+                    output[INTENT], INTENT
+                )
+            if output.get(ACTION_NAME):
+                output[ACTION_NAME] = self._to_sparse_sentence_features(
+                    output[ACTION_NAME], ACTION_NAME
+                )
+
+        # there is always either INTENT or ACTION_NAME
+        name_attribute = next(
+            (
+                attribute
+                for attribute in attributes
+                if attribute in {INTENT, ACTION_NAME}
+            ),
+            None,
+        )
+        if name_attribute and name_attribute not in output:
+            # nlu pipeline didn't create features for user or action
```

Comment:
e.g., when we have `action_name` in the state but it did not get featurized because only character level `CountVectorsFeaturizer` was included in the config.

Reply:
added it to the comment in the code